### PR TITLE
code refactor for controllers registration

### PR DESCRIPTION
**Pull request by zreigz**
**What this PR does / why we need it**: code refactor to use a common pattern for controllers registration

```release-note
NONE
```

### CODE REVIEW

File: api/cmd/user-cluster-controller-manager/main.go [hunk 1]
```diff
@@ -24,6 +26,13 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/runtime/signals"
 )
 
+const metricNamespace = "kubermatic"
+
+// managerMetrics contains managerMetrics that this controller will collect and expose
+type managerMetrics struct {
+	controllers prometheus.Gauge
```
Comment (alvaroaleman) [inline api/cmd/user-cluster-controller-manager/main.go]: Just as a thought regarding this manually created `managerMetrics` that only expose the number of controllers: The manager has a [`MetricsBindAddress` option ](https://godoc.org/sigs.k8s.io/controller-runtime/pkg/manager#Options). Maybe that exposes the same metric and we can just use that and get completely rid of our own metrics struct?

Comment (zreigz) [reply api/cmd/user-cluster-controller-manager/main.go]: good thought, I will check it

Comment (zreigz) [reply api/cmd/user-cluster-controller-manager/main.go]: a lot of useful pieces of information about controllers:
```
# HELP workqueue_adds_total Total number of adds handled by workqueue
# TYPE workqueue_adds_total counter
workqueue_adds_total{name="rbac-user-cluster-controller"} 0
workqueue_adds_total{name="user-cluster-controller"} 0
# HELP workqueue_depth Current depth of workqueue
# TYPE workqueue_depth gauge
workqueue_depth{name="rbac-user-cluster-controller"} 0
workqueue_depth{name="user-cluster-controller"} 0
# HELP workqueue_longest_running_processor_microseconds How many microseconds has the longest running processor for workqueue been running.
# TYPE workqueue_longest_running_processor_microseconds gauge
workqueue_longest_running_processor_microseconds{name="rbac-user-cluster-controller"} 0
workqueue_longest_running_processor_microseconds{name="user-cluster-controller"} 0
# HELP workqueue_queue_latency_seconds How long in seconds an item stays in workqueue before being requested.

```
and much more

**Comment by zreigz** [comment]
@alvaroaleman PTAL

**Comment by alvaroaleman** [comment]
Great change, thanks!

/lgtm
/approve

**Comment by alvaroaleman** [comment]
You can approve yourself for the rbac contorller btw

**Pull request merged**